include/native/reduce.h

```diff
@@ -25,154 +25,65 @@
 #include <functional>
 
 namespace grppi{
-//typename std::enable_if<!is_iterator<Output>::value, bool>::type,
-
-template < typename InputIt, typename Output, typename ReduceOperator>
- typename std::enable_if<!is_iterator<Output>::value, void>::type 
-reduce(parallel_execution_native &p, InputIt first, InputIt last, Output & firstOut, ReduceOperator op) {
-
-    typename ReduceOperator::result_type identityVal = !op(false,true);
 
+template < typename InputIt, typename ReduceOperator>
+typename std::iterator_traits<InputIt>::value_type
+reduce(parallel_execution_native &p, InputIt first, InputIt last, typename std::iterator_traits<InputIt>::value_type init, ReduceOperator && op){
```

Comment:
Type and function object names have been acordingly modified in reduce and map_reduce interfaces.